src/front/wgsl/mod.rs

```diff
@@ -1585,7 +1596,9 @@ impl Parser {
             })
         };
 
-        self.scopes.pop();
+        let span = self.pop_scope(lexer);
+        const_arena.add_span(handle, ArenaSpan::ByteRange(span));
```

Comment:
how would this work in a case where we called `fetch_or_append` and took an existing handle?

Reply:
Not very well, unfortunately, it does indeed stretch the span across :D Thanks, that's a nice catch - I suppose this is a usecase for `set_span_if_unknown` method.